### DISCUSSION TITLE
How to upgrade embedded SQLite?

### DISCUSSION THREAD

**Discussion by c9845** [sqlitebrowser/sqlitebrowser#3206] [Q&A] [answered]
Hello,
I am trying to get involved in this project, mostly to update the embedded SQLite version.  The version included in 3.12.2 is 3.35.5 and is a year and a half old (released April 2021).

Is there a method for updating the SQLite version?  I looked though the source code but cannot determine which file(s) to replace.  I think documenting this process would be helpful since SQLite is updated pretty regularly and thus DB Browser should also be updated pretty regularly.

Thanks!

**Comment by lucydodo** [comment, marked as answer]
The SQLite version is based on the libraries installed on the machine at compile time.
So, To change the SQLite version, install the desired SQLite library and then rebuild DB4S.

In a Windows environment, you can replace the 'sqlite3.dll' file in the DB4S installation path with the latest DLL file.

**Reply by c9845** [reply]
Thanks!

Yes, replacing the .dll on Windows works.  The version displaying in "About" is wrong, but I guess that is expected.

I did find a bug though.  It has to do with STRICT table definitions (https://www.sqlite.org/stricttables.html).  When performing "Modify Table", the columns are not shown and the CREATE TABLE statement is blank.  The note "Warning: There is something wrong with this table definition that our parser doesn't fully understand..." is displayed.  I am using SQLite version 3.39.4.  I will file an issue for this, but this is way outside of my knowledge on how to fix.

**Reply by lucydodo** [reply]
The former has already been reported. (#3135)

The latter is probably because DB4S does not officially support the latest SQLite version,
so the partial implementation may have been less. I will check it out as soon as possible.

**Comment by c9845** [comment]
Yes, I did see that prior issue regarding the incorrect displayed version.  Your explanation makes sense to me and I already assumed that.  Considering very few people will "drop in" a different version of sqlite3 I didn't really expect the the GUI to update!

**Comment by chrisjlocke** [comment]
You can also install the nightly version.
https://nightlies.sqlitebrowser.org/latest/

Download the .zip, extract, and run the .exe.  It doesn't have to be installed, and won't affect the currently installed version.
Replacing just the .dll is not enough - some code within DB4S may have changed to take effect of the changed sqlite version (eg, an updated parser, or bug fix).

**Comment by justinclift** [comment]
Yeah, our nightly builds for (at least) both Windows and macOS generally have a fairly recent version of SQLite.  eg 3.39.4 currently

**Comment by c9845** [comment]
Hi @justinclift,

Took me *way* too long to get back to this, but I did just download the Oct 30th nightly and it does load a database with `STRICT` tables properly.

**Reply by chrisjlocke** [reply]
Thanks for the confirmation. :)  No problem with the delay - just happy it now works.
